pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seedmanagement
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeed represents a Shoot that is registered as Seed.
+type ManagedSeed struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the ManagedSeed.
+	Spec ManagedSeedSpec
+	// Most recently observed status of the ManagedSeed.
+	Status ManagedSeedStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeedList is a list of ManagedSeed objects.
+type ManagedSeedList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	metav1.ListMeta
+	// Items is the list of ManagedSeeds.
+	Items []ManagedSeed
+}
+
+// ManagedSeedSpec is the specification of a ManagedSeed.
+type ManagedSeedSpec struct {
+	// Shoot is the Shoot that will be registered as Seed.
+	Shoot *Shoot
+	// Seed describes the Seed that will be registered.
+	// Either Seed or Gardenlet must be specified.
+	Seed *SeedTemplateSpec
+	// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+	Gardenlet *Gardenlet
+}
+
+// Shoot identifies the Shoot that will be registered as Seed.
```

Comment:
OK

Reply:
Is there a guideline somewhere (for Kubernetes or specifically for Gardener) regarding how such comments should be phrased? E.g. using "the" vs. "a" or "will" vs. "should"? 

Without such a guideline, I find some of your proposals quite opinionated. I don't mind following your suggestions, but I sometimes have no clue why do you think they are better, as in the case above. This also means I will very likely end up with similar "wrong" comments in the future that you would feel obliged to correct, and I don't really like this. 

To break down the simple comment above: "identifies **the** Shoot" is better IMO because there is a specific, single shoot that needs to be identified. Between "will be", "should be" and "is" the difference is largely a matter of taste here, but of course it's a good idea to keep it consistent. I mostly used "will be", you decided to correct in a few places to "should  be".

Reply:
I'm not aware of such a guideline regarding API documentation.

Actually, I'm sorry about this particular suggestion. I think I wanted to remark something else originally, but changed it again later on. I also lost a bit track in this review. Yes, this comment is fine and doesn't need to be changed.

In general, please see minor suggestions like this one as "nit comments" and not "change requests". I just wanted to propose alternative phrasing in some places that would be more intuitive to me personally when reading the API docs.
Though, I got your point, this isn't really helpful and rather hindering sometimes (especially in large PRs like this one), and I will avoid such suggestions in the future.